sdcm/cluster_baremetal.py

```diff
@@ -74,13 +98,20 @@ def __init__(self, **kwargs):
         self.nodes = []
         self.credentials = kwargs.pop('credentials')
         n_nodes = kwargs.get('n_nodes')
-        self._node_public_ips = kwargs.get('public_ips', None) or []
-        self._node_private_ips = kwargs.get('private_ips', None) or []
+        self._node_public_ips = kwargs.pop('public_ips', None) or []
+        self._node_private_ips = kwargs.pop('private_ips', None) or []
         node_cnt = n_nodes[0] if isinstance(n_nodes, list) else n_nodes
         if len(self._node_public_ips) < node_cnt or len(self._node_private_ips) < node_cnt:
             raise NodeIpsNotConfiguredError('Physical hosts IPs are not configured!')
         super().__init__(**kwargs)
 
+    @property
+    def ssh_username(self) -> str:
```

Comment:
I don't understand the need of this property, also we don't want default like `ubuntu` to be in the code.
this should have it's own new sct configuration option

Reply:
I've moved this into bare metal config already, maybe just raise an exception there if it's unset?

Reply:
Changed to re-raise AttributeError

Reply:
Why do you need to catch exception at all in this case?